packages/ui-components/src/components/Inputs/FileUploadField.js

```diff
@@ -0,0 +1,64 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { GreyButton } from '../Button';
+import { FlexStart } from '../Layout';
+import { SaveAlt } from '../Icons';
+
+const HiddenFileInput = styled.input`
+  width: 0.1px;
+  height: 0.1px;
+  opacity: 0;
+  overflow: hidden;
+  position: absolute;
+  z-index: -1;
+`;
+
+const FileName = styled.span`
+  font-size: 1rem;
+  color: ${props => props.theme.palette.text.secondary};
+  margin-left: 0.8rem;
+`;
+
+export const FileUploadField = ({ onChange, name }) => {
+  const inputEl = useRef(null);
+  const [fileName, setFileName] = useState(null);
+
+  const handleChange = event => {
+    let newName;
+    const input = inputEl.current;
+
+    if (input.files && input.files.length > 1) {
+      newName = `${input.files.length} files selected`;
+    } else {
+      newName = event.target.value.split('\\').pop();
```

Comment:
That's a clever way of getting the last item of an array when we don't care whether it's mutated!